ai/core/src/unitycatalog/ai/core/utils/callable_utils.py

```diff
@@ -345,6 +346,28 @@ def process_parameter(
         return f"{param_name} {sql_type} COMMENT '{param_comment}'"
 
 
+def construct_dependency_statement(
+    dependencies: Optional[list[str]] = None, environment_version: str = "None"
+) -> str:
+    """
+    Constructs the dependency statement for the SQL function.
+
+    Args:
+        dependencies: An optional list of PyPI dependencies for the function to utilize in the execution environment
+        environment_version: The version of the environment to use for the function. Defaults to "None".
+
+    Returns:
+        str: The constructed dependency statement.
+    """
+    if dependencies:
+        dependencies_json = json.dumps(dependencies)
+        return f"\nENVIRONMENT (dependencies = '{dependencies_json}', environment_version = '{environment_version}')"
```

Comment:
Do we need environment_version if it's None?

Reply:
Yes, it errors out if a dependency is specified without the environment argument being supplied. This is to promote base image reuse in the backend.